sdk/cpp/core/src/gnmi_client.cpp

```diff
@@ -0,0 +1,273 @@
+#include <memory>
+#include <string>
+#include <sstream>
+#include <vector>
+#include <iterator>
+
+#include "errors.hpp"
+#include "gnmi_client.hpp"
+#include "types.hpp"
+#include "logger.hpp"
+#include "path_api.hpp"
+#include "json.hpp"
+
+#include <libyang/libyang.h>
+#include <grpc++/grpc++.h>
+#include <grpc++/client_context.h>
+#include <grpc++/create_channel.h>
+#include <libgnmi/gnmi.grpc.pb.h>
+#include <libgnmi/gnmi.pb.h>
+#include "entity_data_node_walker.hpp"
+#include "ietf_parser.hpp"
+#include "ydk_yang.hpp"
+
+using grpc::Channel;
+using grpc::ClientContext;
+using grpc::Status;
+using gnmi::gNMI;
+using gnmi::CapabilityRequest;
+using gnmi::CapabilityResponse;
+using gnmi::GetRequest;
+using gnmi::GetResponse;
+
+using namespace std;
+using namespace ydk;
+using json = nlohmann::json;
+
+typedef vector<string> StringVec;
+std::ifstream capfile("/usr/local/share/ydk/0.0.0.0\:50051/capabilities.txt");
+
+gNMIClient::gNMIClient(shared_ptr<Channel> channel)
+	: stub_(gNMI::NewStub(channel)){}
+
+StringVec gNMIClient::Capabilities() 
+{
+	CapabilityRequest request;
+	CapabilityResponse response;
+	gNMIClient::get_capabilities(request, &response);
+	return capabilities;
+}
+
+int gNMIClient::connect() 
+{
+    grpc::CreateChannel("0.0.0.0:50051", grpc::InsecureChannelCredentials());
+    init_capabilities();
+    return EXIT_SUCCESS;
+}
+
+void gNMIClient::init_capabilities() 
+{
+	std::string cap;
+	while (std::getline(capfile, cap)) 
+	{
+		capabilities.push_back(cap);
+		//std::cout << cap;
+	}
+}
+
+bool gNMIClient::get_capabilities(const CapabilityRequest& request, CapabilityResponse* response) 
+{
+	grpc::ClientContext context;
+	grpc::Status status = stub_->Capabilities(&context, request, response);
+	if (!status.ok()) 
+	{
+	  std::cout << "Capabilities RPC Failed." << std::endl;
+	  return false;
+	}
+	if (!response->gnmi_version().size() > 0) 
+	{
+	  std::cout << "Server has not returned gnmi version" << std::endl;
+	  return false;
+	} else 
+	{
+		std::cout << "Receive capability response with gNMI version: " << response->gnmi_version() << std::endl;
+		for (unsigned int i = 0, n = response->supported_models_size(); i < n; i++) 
+		{
+			std::cout << "supported_models size" << response->supported_models_size() << std::endl;
+			std::cout << "supported_models";
+			response->supported_models(i);
+		}
+		for (int i = 0, n = response->supported_encodings_size(); i < n; i++) 
+		{
+			std::cout << "supported_models size" << response->supported_encodings_size() << std::endl;
+			std::cout << "supported_models";
+			response->supported_encodings(i);
+		}
+	}
+    return true;
+}
+
+void gNMIClient::Get() {}
+
+GetRequest gNMIClient::ParseGetRequest(GetRequest request, string payload) 
+{	
+	::gnmi::Path* prefix = new ::gnmi::Path();
+	::gnmi::Path* path;
+
+	// create parsable payload 
+	std::string payload_recieved = "{";
+	payload_recieved.append(payload);
+	payload_recieved.append("}");
+	std::cout << payload << endl;
```

Comment:
Like Xiaoqin mentioned, we should probably be using the ydk logging functions instead of `cout`